File: mysql-async/src/main/scala/com/github/mauricio/async/db/mysql/message/server/ServerMessage.scala
/*
 * Copyright 2013 Maurício Linhares
 *
 * Maurício Linhares licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 *
 */

package com.github.mauricio.async.db.mysql.message.server

import com.github.mauricio.async.db.KindedMessage

object ServerMessage {

  final val ServerProtocolVersion = 10
  final val Error = -1
  final val Ok = 0
  final val EOF = -2

  // these messages don't actually exist
  // but we use them to simplify the switch statements
  final val ColumnDefinition = 100
  final val ColumnDefinitionFinished = 101
  final val ParamProcessingFinished = 102
  final val ParamAndColumnProcessingFinished = 103
  final val Row = 104
  final val BinaryRow = 105
  final val PreparedStatementPrepareResponse = 106

}

class ServerMessage( val kind : Int ) extends KindedMessage

File: mysql-async/src/main/java/com/github/jasync/sql/db/mysql/message/server/ServerMessage.kt
package com.github.jasync.sql.db.mysql.message.server

import com.github.jasync.sql.db.KindedMessage


abstract class ServerMessage(override val kind: Int) : KindedMessage {

    companion object {
        const val ServerProtocolVersion = 10
        const val Error = -1
        const val Ok = 0
        const val EOF = -2

        // these messages don't actually exist
        // but we use them to simplify the switch statements
        const val ColumnDefinition = 100
        const val ColumnDefinitionFinished = 101
        const val ParamProcessingFinished = 102
        const val ParamAndColumnProcessingFinished = 103
        const val Row = 104
        const val BinaryRow = 105
        const val PreparedStatementPrepareResponse = 106
    }
}
